### ISSUE TITLE
Key Error with unclear source

### ISSUE THREAD

**Issue by fisseb4** [JuBiotech/bletl#24]
Hi guys,

running the parse of my biolector file I get an error and I am not sure where this is coming from (you can see the error below). I am quite sure that this is regarding my input csv. I exported this from the biolector twice but therer error is still the same. The csv from a different run works totally fine with the script. Both are with the same biolector software and bletl version analyzed (or tried to analyze for the one with the error. Based on the error, do you have any clue where the error is coming from? I also tried to delete all or parts of the results (after line 2838 in my csv) because I thought that there may be corrupt symbols inside. But this only changes the error numer from '05' to '01' or '02'. Do you have any clue?
Thank you for your help in advance!

```
KeyError                                  Traceback (most recent call last)
~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\base.py in get_loc(self, key, method, tolerance)
   3620             try:
-> 3621                 return self._engine.get_loc(casted_key)
   3622             except KeyError as err:

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\_libs\index.pyx in pandas._libs.index.IndexEngine.get_loc()

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\_libs\index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas\_libs\hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

pandas\_libs\hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

KeyError: '05'

The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_14564\2706581641.py in <cell line: 3>()
      1 # now provide the filepath
      2 filepath = pathlib.Path(data_raw)
----> 3 bldata = bletl.parse(filepath)
      4 
      5 # example with new temerature and lot-number:

~\AppData\Local\Programs\Python\Python310\lib\site-packages\bletl\core.py in parse(filepaths, drop_incomplete_cycles, lot_number, temp, cal_0, cal_100, phi_min, phi_max, pH_0, dpH)
    234         data.metadata['date_end'] = fragments[-1].metadata['date_end']
    235     else:
--> 236         data = _parse(filepaths, drop_incomplete_cycles, lot_number, temp, cal_0, cal_100, phi_min, phi_max, pH_0, dpH)
    237 
    238     return data

~\AppData\Local\Programs\Python\Python310\lib\site-packages\bletl\core.py in _parse(filepath, drop_incomplete_cycles, lot_number, temp, cal_0, cal_100, phi_min, phi_max, pH_0, dpH)
    127     """
    128     parser = get_parser(filepath)
--> 129     data = parser.parse(filepath, lot_number, temp, cal_0, cal_100, phi_min, phi_max, pH_0, dpH)
    130 
    131     if (not data.measurements.empty) and drop_incomplete_cycles:

~\AppData\Local\Programs\Python\Python310\lib\site-packages\bletl\parsing\blpro.py in parse(self, filepath, lot_number, temp, cal_0, cal_100, phi_min, phi_max, pH_0, dpH)
     67                     bld[key] = fts
     68         else:
---> 69             for key, fts in transform_into_filtertimeseries(bld.metadata, bld.measurements, bld.filtersets, False):
     70                 bld[key] = fts
     71 

~\AppData\Local\Programs\Python\Python310\lib\site-packages\bletl\parsing\blpro.py in transform_into_filtertimeseries(metadata, measurements, filtersets, return_uncalibrated_optode_data)
    401         elif fs.filter_type == 'Intensity':
    402             key = fs.filter_name
--> 403             times = measurements.xs(filter_number, level='filterset')['time'].unstack()
    404             values = measurements.xs(filter_number, level='filterset')['amp_ref_1'].unstack()
    405         else:

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\generic.py in xs(self, key, axis, level, drop_level)
   3834             if not isinstance(labels, MultiIndex):
   3835                 raise TypeError("Index must be a MultiIndex")
-> 3836             loc, new_ax = labels.get_loc_level(key, level=level, drop_level=drop_level)
   3837 
   3838             # create the tuple of the indexer

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\multi.py in get_loc_level(self, key, level, drop_level)
   2970             level = [self._get_level_number(lev) for lev in level]
   2971 
-> 2972         loc, mi = self._get_loc_level(key, level=level)
   2973         if not drop_level:
   2974             if lib.is_integer(loc):

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\multi.py in _get_loc_level(self, key, level)
   3111                 return indexer, maybe_mi_droplevels(indexer, ilevels)
   3112         else:
-> 3113             indexer = self._get_level_indexer(key, level=level)
   3114             if (
   3115                 isinstance(key, str)

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\multi.py in _get_level_indexer(self, key, level, indexer)
   3220         else:
   3221 
-> 3222             idx = self._get_loc_single_level_index(level_index, key)
   3223 
   3224             if level > 0 or self._lexsort_depth == 0:

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\multi.py in _get_loc_single_level_index(self, level_index, key)
   2800             return -1
   2801         else:
-> 2802             return level_index.get_loc(key)
   2803 
   2804     def get_loc(self, key, method=None):

~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\indexes\base.py in get_loc(self, key, method, tolerance)
   3621                 return self._engine.get_loc(casted_key)
   3622             except KeyError as err:
-> 3623                 raise KeyError(key) from err
   3624             except TypeError:
   3625                 # If we have a listlike key, _check_indexing_error will raise

KeyError: '05'
```

**Comment by michaelosthege** [comment]
@fisseb4 thank you for raising the issue. @BioAutomation forwarded me your CSV which I anonymized so we can use it in the unit tests. I hope that's ok?

It's not a fix, but in #25 I added a test case that reproduces the issue.

To me it looks like one of the fluorescence filtersets was `REFOVERLD` in all cycles and was therefore dropped. This lead to the `KeyError` at a later point in the processing.
